### PR TITLE
Prototype runner

### PR DESCRIPTION
**Pull request by toyamarinyon**
> [!NOTE]
> This change is a prototype for flow execution. Since we cannot yet verify its operation, we will release it in private beta status.

> [!NOTE]
> After merging, please perform schema and data migration. For details, please check the "Release Flow:" section.

This pull request introduces a new flow management system to the playground application, including actions, reducers, and UI components to handle flow state and execution. The most important changes include adding new types and actions for flow management, implementing a reducer to handle flow actions, creating a run button component, and updating the header and viewer components to integrate the new flow functionalities.

### Flow Management System:

* `app/(playground)/p/[agentId]/beta-proto/flow/action.ts`: Added new types and actions for managing flow state, including `V2FlowAction`, `SetFlowAction`, and `v2FlowReducer`. ([app/(playground)/p/[agentId]/beta-proto/flow/action.tsR1-R42](diffhunk://#diff-d396c5644caf1a3ccccea9e15e670796f688d0984ab9e6d8c5ba43b10d0d7eecR1-R42))
* `app/(playground)/p/[agentId]/beta-proto/flow/types.ts`: Defined new types for flow management, such as `Flow`, `QueuedFlow`, `RunningFlow`, and their statuses. ([app/(playground)/p/[agentId]/beta-proto/flow/types.tsR1-R76](diffhunk://#diff-3cf29eee6c0733e7c9750cb244af18e5c8f52f1a3d2cf5caba9c0191ec0e5052R1-R76))

### Reducer and Context Updates:

* `app/(playground)/p/[agentId]/beta-proto/graph/reducer.ts`: Updated the graph reducer to handle flow actions by integrating `v2FlowReducer`. ([app/(playground)/p/[agentId]/beta-proto/graph/reducer.tsR1](diffhunk://#diff-6eb17a4767e3d461dba850c4e45c6885cd1bf909f297845e85f46b0c66ae565eR1), [app/(playground)/p/[agentId]/beta-proto/graph/reducer.tsR29-R37](diffhunk://#diff-6eb17a4767e3d461dba850c4e45c6885cd1bf909f297845e85f46b0c66ae565eR29-R37))
* `app/(playground)/p/[agentId]/beta-proto/graph/types.ts`: Added `flow` to the `Graph` type to include flow state in the graph context. ([app/(playground)/p/[agentId]/beta-proto/graph/types.tsR3-R5](diffhunk://#diff-b03e6f0c4a5ac79fc4727ec410f9f990b12dd9bcb61cda1ef7bd1753111a48afR3-R5), [app/(playground)/p/[agentId]/beta-proto/graph/types.tsR15-R21](diffhunk://#diff-b03e6f0c4a5ac79fc4727ec410f9f990b12dd9bcb61cda1ef7bd1753111a48afR15-R21))

### UI Components:

* `app/(playground)/p/[agentId]/beta-proto/flow/components/run-button.tsx`: Created a new `RunButton` component to trigger flow execution, with support for single and multiple final nodes. ([app/(playground)/p/[agentId]/beta-proto/flow/components/run-button.tsxR1-R76](diffhunk://#diff-91d3bbb0a83d8d84220f34bb35ce0d2f04e841f121b073326ff6b557fb3ee3f7R1-R76))
* `app/(playground)/p/[agentId]/beta-proto/header.tsx`: Replaced the static run button with the new `RunButton` component. ([app/(playground)/p/[agentId]/beta-proto/header.tsxR6](diffhunk://#diff-d6225e27c2fc7d5de30daba1402d59f89b3825468a20c6631705351754b063efR6), [app/(playground)/p/[agentId]/beta-proto/header.tsxL43-R44](diffhunk://#diff-d6225e27c2fc7d5de30daba1402d59f89b3825468a20c6631705351754b063efL43-R44))
* `app/(playground)/p/[agentId]/beta-proto/viewer.tsx`: Updated the viewer component to display the flow state or a message if no flow exists. ([app/(playground)/p/[agentId]/beta-proto/viewer.tsxR1-R13](diffhunk://#diff-f21fcf2b33e3dd1c7959cb42f2f615400f0a1ca2c260a2939055dfc8cdb90fe1R1-R13), [app/(playground)/p/[agentId]/beta-proto/viewer.tsxR27-R36](diffhunk://#diff-f21fcf2b33e3dd1c7959cb42f2f615400f0a1ca2c260a2939055dfc8cdb90fe1R27-R36), [app/(playground)/p/[agentId]/beta-proto/viewer.tsxR47-R49](diffhunk://#diff-f21fcf2b33e3dd1c7959cb42f2f615400f0a1ca2c260a2939055dfc8cdb90fe1R47-R49))

### Flow Execution:

* `app/(playground)/p/[agentId]/beta-proto/flow/composite-actions.ts`: Implemented the `runFlow` action to handle the flow execution process, including setting flow state and dispatching actions. ([app/(playground)/p/[agentId]/beta-proto/flow/composite-actions.tsR1-R54](diffhunk://#diff-20d9584da2d54a3aa33b1c809934293abf3a79d1ff79b24193478628bd8c8278R1-R54))
* `app/(playground)/p/[agentId]/beta-proto/flow/server-action.ts`: Added server-side logic for running actions within a flow, including placeholder methods for text generation and web search. ([app/(playground)/p/[agentId]/beta-proto/flow/server-action.tsR1-R159](diffhunk://#diff-04e6693847b97a9e955c52afb29f78eab6cb8f07c64c0293bf4a7fba1e8cf096R1-R159))

These changes collectively introduce a robust flow management system, enhancing the application's ability to manage and execute complex workflows.

#### Migrate schema

```sh
 bun drizzle-kit migrate
```

#### Migrate data
```sh
bun scripts/20241024-db-patch-add-agent-id.ts
```

### CODE REVIEW

File: app/(playground)/p/[agentId]/beta-proto/graph/provider.tsx [hunk 1]
```diff
@@ -52,7 +52,9 @@ export const GraphProvider: FC<PropsWithChildren<GraphProviderProps>> = ({
 	useEffect(() => {
 		if (isInitialMount.current) {
 			isInitialMount.current = false;
+			console.log({ a: state.graph });
```
Comment (shige) [inline app/(playground)/p/[agentId]/beta-proto/graph/provider.tsx]: nits: is this necessary?

File: app/(playground)/p/[agentId]/beta-proto/graph/provider.tsx [hunk 2]
```diff
@@ -52,7 +52,9 @@ export const GraphProvider: FC<PropsWithChildren<GraphProviderProps>> = ({
...
 		} else {
+			console.log({ b: state.graph });
```
Comment (shige) [inline app/(playground)/p/[agentId]/beta-proto/graph/provider.tsx]: nits: is this necessary?

File: app/(playground)/p/[agentId]/page.tsx [hunk 3]
```diff
@@ -47,7 +47,7 @@ export default async function AgentPlaygroundPage({
 				uploadFileToPromptNodeFlag,
 				webSearchNodeFlag,
 				debugFlag,
-				viewFlag,
+				viewFlag: true,
```
Comment (shige) [inline app/(playground)/p/[agentId]/page.tsx]: @toyamarinyon ❓ 

Did they forget to remove it off? 
If they didn't forget to turn it off, I would like them to delete the feature flag itself, but I understand that the feature implementation is not yet complete.

**Review by shige — COMMENTED** [review]
I added comments on the points I wanted to confirm.  
Please check them!

File: app/(playground)/p/[agentId]/page.tsx (hunk 3 above)
Comment (toyamarinyon) [reply app/(playground)/p/[agentId]/page.tsx]: Sorry, I made a commit by mistake.
Fixed in commit 7c47322642d0028b8bc7cf59204b414b49663311

File: app/(playground)/p/[agentId]/beta-proto/graph/provider.tsx (hunk 2 above)
Comment (toyamarinyon) [reply app/(playground)/p/[agentId]/beta-proto/graph/provider.tsx]: Sorry, I made a commit by mistake.
Fixed in commit ea2d0311e246e323e2a1bde7ab12ef33ba1ffab0

**Comment by toyamarinyon** [comment]
@shige I forgot to remove the debugging code. I've removed it now, please review again.

**Review by shige — APPROVED** [review]
LGTM 💯 
Thank you for fixing it right away!

Tomorrow morning, I'll deploy, migrate, and update the database. 📆

**Comment by shige** [comment]
FYI: Released 🚀  @toyamarinyon 
- [x] deployed to production
- [x] `bun drizzle-kit migrate`
- [x] `bun scripts/20241024-db-patch-add-agent-id.ts`
```
Starting the update of agent data to set 'agentId' property on graph.
Updating 73 agents...
  ├ Processing chunk 1/8 (10 agents)
  │ └ Completed chunk 1/8
  ├ Processing chunk 2/8 (10 agents)
  │ └ Completed chunk 2/8
  ├ Processing chunk 3/8 (10 agents)
  │ └ Completed chunk 3/8
  ├ Processing chunk 4/8 (10 agents)
  │ └ Completed chunk 4/8
  ├ Processing chunk 5/8 (10 agents)
  │ └ Completed chunk 5/8
  ├ Processing chunk 6/8 (10 agents)
  │ └ Completed chunk 6/8
  ├ Processing chunk 7/8 (10 agents)
  │ └ Completed chunk 7/8
  ├ Processing chunk 8/8 (3 agents)
  │ └ Completed chunk 8/8
All agents have been successfully updated!
```
- [x] checked the operations in production

**Pull request merged**